### ISSUE TITLE
[Range Presplitting of tablets] [Backup and Restore][DDL] Restore backup fails in the import_snapshot step with OBJECT_NOT_FOUND (master error 3)

### ISSUE THREAD

**Issue by kripasreenivasan** [yugabyte/yugabyte-db#14825]
Jira Link: [DB-4131](https://yugabyte.atlassian.net/browse/DB-4131)
### Description

Scenario attempted on build 2.17.1.0-b138

**Steps to reproduce the issue:**

1. Create a universe with packed columns disabled
2. Create a range sharded table in database `range_sharded_db`
3. Trigger a workload to populate the table.
4. Trigger a rolling restart on the universe.
5. Ensure the workload continues
6. Once the gflag edit completes, create a backup of the database- BKUP001
7. Create a PITR snapshot schedule (Note- No restores executed, Adding this line for documentation sake)
8. Execute alter table DDLs' on the table and change the schema a bit- I added 5 new columns
9. Insert more data into the table
10. Create a backup of the database- BKUP002.
11. Restore the backup created pre-DDL execution - BKUP001 on to a new database (Restore on the same database is not possible owing to the PITR schedule)
12. Verify successful restore
13. Restore the backup- BKUP002- taken post DDL execution- on to a new database
**Issue:** Restore backup of the range_sharded table fails in the import_snapshot step with OBJECT_NOT_FOUND (master error 3)
```
Error running import_snapshot: Invalid argument (yb/master/catalog_manager_ent.cc:2226): Unable to import snapshot meta file /tmp/yb_backup_umpuxyumseeaqlkj/SnapshotInfoPB: YSQL table not found: customers_rs: OBJECT_NOT_FOUND (master error 3)
```

Error observed on doing  a YSQLDump:
```
de_set_next_pg_type_oid('16416'::pg_catalog.oid);
ysqlsh:/tmp/YSQLDump_test:75: ERROR:  syntax error at or near "de_set_next_pg_type_oid"
LINE 19: de_set_next_pg_type_oid('16416'::pg_catalog.oid);
         ^
-- For binary upgrade, must preserve pg_type array oid
SELECT pg_catalog.binary_upgrade_set_next_array_pg_type_oid('16415'::pg_catalog.oid);
 binary_upgrade_set_next_array_pg_type_oid 
```

Sequence of steps from logs:
```
YW 2022-11-03T10:27:17.360Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - 2022-11-03 10:25:44,230 INFO: Completed phase 0: START [Time taken for phase: 0:00:05.991273]
YW 2022-11-03T10:27:17.360Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - 2022-11-03 10:25:44,230 INFO: [app] Starting phase 1: Create objects via YSQL dumps
YW 2022-11-03T10:27:17.360Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - 2022-11-03 10:25:44,230 INFO: [app] Create YSQL tables from /tmp/yb_backup_umpuxyumseeaqlkj/YSQLDump
YW 2022-11-03T10:27:17.360Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - 2022-11-03 10:25:44,244 INFO: [app] Renaming YSQL DB from 'range_sharding' into 'range_sharding_B2'
YW 2022-11-03T10:27:17.360Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - 2022-11-03 10:25:46,212 INFO: Completed phase 1: Create objects via YSQL dumps [Time taken for phase: 0:00:01.982364]
YW 2022-11-03T10:27:17.360Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - 2022-11-03 10:25:46,212 INFO: [app] Starting phase 2: Import snapshot
YW 2022-11-03T10:27:17.360Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - 2022-11-03 10:25:46,286 ERROR: Failed to run command [[ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ControlMaster=auto -o 'ControlPath=~/.ssh/ssh-%r@%h:%p' -o ControlPersist=1m -i /opt/yugabyte/yugaware/data/keys/08c0ba0e-3558-40fc-94e5-a87a627de8c5/yb-15-aws-portal-1-key.pem -p 54422 -q yugabyte@172.151.32.143 'cd / &&  bash -c '"'"'/home/yugabyte/master/bin/yb-admin --master_addresses 172.151.28.185:7100,172.151.32.143:7100,172.151.57.100:7100 import_snapshot /tmp/yb_backup_umpuxyumseeaqlkj/SnapshotInfoPB range_sharding_B2 --never_fsync=false'"'"'' ]]: code=1 output=Read snapshot meta file /tmp/yb_backup_umpuxyumseeaqlkj/SnapshotInfoPB
YW 2022-11-03T10:27:17.360Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - Importing snapshot ff406b2b-e5d0-4df7-8463-9729ee95118e (COMPLETE)
YW 2022-11-03T10:27:17.360Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - Target imported YSQL database name: range_sharding_B2
YW 2022-11-03T10:27:17.360Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - YSQL database being imported: range_sharding
YW 2022-11-03T10:27:17.360Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - Table type: table
YW 2022-11-03T10:27:17.360Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - Target imported table name: range_sharding_B2.customers_rs
YW 2022-11-03T10:27:17.360Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - Table being imported: range_sharding.customers_rs
YW 2022-11-03T10:27:17.360Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - Error running import_snapshot: Invalid argument (yb/master/catalog_manager_ent.cc:2226): Unable to import snapshot meta file /tmp/yb_backup_umpuxyumseeaqlkj/SnapshotInfoPB: YSQL table not found: customers_rs: OBJECT_NOT_FOUND (master error 3)
YW 2022-11-03T10:27:17.361Z [DEBUG] 87851be1-7489-4351-b2f6-a1077db824db from ShellProcessHandler in TaskPool-RestoreBackup(f07348ae-6221-4e1d-ad18-895903c9eb78)-0 - Usage: /home/yugabyte/master/bin/yb-admin import_snapshot  <file_name> [<namespace> <table_name> [<table_name>]...]
```
Schema used:
```
Range Sharded table-

CREATE TABLE customers_rs (
    id bigserial NOT NULL, 
    customer_id int NOT NULL,
    company_name character varying(40) NOT NULL,
    contact_name character varying(30),
    contact_title character varying(30),
    address character varying(60),
    city character varying(15),
    region character varying(15),
    postal_code character varying(10),
    country character varying(15),
    phone character varying(24),
    fax character varying(24),
    PRIMARY KEY (customer_id ASC)
) SPLIT AT VALUES ((1000), (5000), (10000), (15000), (20000), (25000), (30000), (35000), (55000), (85000), (110000), (150000),(250000), (300000), (350000), (400000), (450000), (500000), (1000000)) ;


DO
$do$
BEGIN 
   FOR i IN 1..200000 LOOP
      INSERT INTO customers_rs(id, customer_id, company_name, contact_name, contact_title, address, city, region, postal_code, country, phone, fax,new_col_1,new_col_2,new_col_3) VALUES (i, i, 'Yugabyte', 'Kripa', 'Sreenivasan','Wellington street', 'Blr', 'KN', '560085', 'India', '9976523541', '8642333871','data_1','data_2','data_3');
   END LOOP;
END
$do$;

```

CC: @bmatican , @rthallamko3

**Comment by kripasreenivasan** [comment]
This scenario works in 2.14 and fails in 2.16

**Comment by yifanguan** [comment]
Issue resolved by commit: ed3cd2d

**Comment by kripasreenivasan** [comment]
Fix verified.
